### PR TITLE
Extract attributes also in local logs

### PR DESCRIPTION
**Pull request by tagomoris**
For example:
```
15:05:00.913 [XNIO-1 task-1] INFO com.treasuredata.bigdam.pool.http.resources.PrivateApiForTestsResource - yaaaaaaaaaaaay {k1=v1, list=[vvv, foo, bar]}
```

### CODE REVIEW

**Comment by tagomoris** [comment]
@komamitsu Could you review this change?

File: src/main/java/com/treasuredata/bigdam/log/Log.java [hunk 1]
```diff
@@ -344,6 +342,21 @@ else if (attributeKeysMasked.contains(key)) {
...
+    private Map<String, Object> buildEvent(final String messageKey, final String message, final Throwable e, final Map<String, ? extends Object> attrs)
+    {
+        HashMap<String, Object> event = new HashMap<>();
+        if (messageKey != null) {
+            event.put(messageKey, message);
+        }
+        if (e != null) {
+            event.put("errorClass", e.getClass().getName());
+            event.put("error", e.getMessage());
+        }
```
Comment (komamitsu) [inline src/main/java/com/treasuredata/bigdam/log/Log.java:357]: This information is sent to Sentry or something, right? Why don't you add a stack trace as well? It's really useful for investigation.

File: src/main/java/com/treasuredata/bigdam/log/Log.java [hunk 2]
```diff
@@ -298,17 +300,13 @@ private EventTime eventTime(final Instant now)
...
-        }
-        if (e != null) {
-            event.put("errorClass", e.getClass().getName());
-            event.put("error", e.getMessage());
+        if (attrs == null) {
+            return ImmutableMap.of();
         }
-        event.putAll(defaultAttributes);
+
+        HashMap<String, Object> event = new HashMap<>();
```
Comment (komamitsu) [inline src/main/java/com/treasuredata/bigdam/log/Log.java]: Could you add a comment why you put all entries to another Map?

Comment (komamitsu) [inline src/main/java/com/treasuredata/bigdam/log/Log.java]: You can define the variable as `Map` not `HashMap`.

**Review by komamitsu — APPROVED** [review]
LGTM 👍 
I left some comments, though.

File: src/main/java/com/treasuredata/bigdam/log/Log.java (hunk 2 above)
Comment (tagomoris) [reply src/main/java/com/treasuredata/bigdam/log/Log.java]: OK.

Comment (tagomoris) [reply src/main/java/com/treasuredata/bigdam/log/Log.java]: Right.

File: src/main/java/com/treasuredata/bigdam/log/Log.java [hunk 3]
```diff
@@ -344,6 +342,21 @@ else if (attributeKeysMasked.contains(key)) {
...
 
+    private Map<String, Object> buildEvent(final String messageKey, final String message, final Throwable e, final Map<String, ? extends Object> attrs)
+    {
+        HashMap<String, Object> event = new HashMap<>();
+        if (messageKey != null) {
+            event.put(messageKey, message);
+        }
+        if (e != null) {
+            event.put("errorClass", e.getClass().getName());
+            event.put("error", e.getMessage());
```
Comment (komamitsu) [inline src/main/java/com/treasuredata/bigdam/log/Log.java:356]: Ah, I noticed the cause of this exception won't be sent. How about sending a cause recursively if it exists?

File: src/main/java/com/treasuredata/bigdam/log/Log.java (hunk 1 above)
Comment (tagomoris) [reply src/main/java/com/treasuredata/bigdam/log/Log.java:357]: No. Information sent to Sentry is build in `sendException(Throwable, Map)` method.
That method handles `Throwable e` directly, including stack traces. `buildEvent` does nothing about Sentry.

Comment (komamitsu) [reply src/main/java/com/treasuredata/bigdam/log/Log.java:357]: 👍

File: src/main/java/com/treasuredata/bigdam/log/Log.java (hunk 3 above)
Comment (tagomoris) [reply src/main/java/com/treasuredata/bigdam/log/Log.java:356]: Do you mean to add `causeClass` and `cause` fields?

Comment (tagomoris) [reply src/main/java/com/treasuredata/bigdam/log/Log.java:356]: Hmm, it's not recursive way. How can do it?

Comment (komamitsu) [reply src/main/java/com/treasuredata/bigdam/log/Log.java:356]: > Do you mean to add causeClass and cause fields?

Right. Also, `List` can represent nested multiple causes, I think.

However, I thought this information would be sent to Sentry. So adding cause exceptions here seemed great. According to your earlier comment, it's not sent to Sentry and maybe we don't need to do that.

Comment (tagomoris) [reply src/main/java/com/treasuredata/bigdam/log/Log.java:356]: I see. I agree that we can refer Sentry dashboard when we want to investigate about errors in details. I'll skip cause exceptions for now.

**Comment by tagomoris** [comment]
Merged. Thank you!

**Pull request merged**